Source/rclUE/Private/ROS2Node.cpp

```diff
@@ -167,55 +166,83 @@ void AROS2Node::AddService(const FString ServiceName, const TSubclassOf<UROS2Gen
     }
 }
 
-void AROS2Node::AddPublisher(UROS2Publisher* Publisher)
+void AROS2Node::AddPublisher(UROS2Publisher* InPublisher)
 {
-    check(IsValid(Publisher));
+    check(IsValid(InPublisher));
 
-    if (!Publisher->UpdateDelegate.IsBound())
+    if (!InPublisher->UpdateDelegate.IsBound())
     {
-        UE_LOG(LogROS2Node, Warning, TEXT("UpdateDelegate is not set - is this on purpose? (%s)"), *__LOG_INFO__);
+        UE_LOG(LogROS2Node, Warning, TEXT("[%s] UpdateDelegate is not set - is this on purpose? (%s)"), *GetName(), *__LOG_INFO__);
     }
 
-    Publisher->RegisterComponent();
-    Publisher->OwnerNode = this;
-    Publishers.Add(Publisher);
+    if (false == Publishers.Contains(InPublisher))
+    {
+        InPublisher->RegisterComponent();
+        InPublisher->OwnerNode = this;
+        Publishers.Add(InPublisher);
+    }
+    else
+    {
+        UE_LOG(LogROS2Node, Warning, TEXT("[%s] Publisher is re-added (%s)"), *GetName(), *__LOG_INFO__);
```

Comment:
sound like new publisher overwirte existing one.
it maybe better to overwrite, I guess?

Reply:
Well, both `Publisher` and `ROS2Node` are just pointers handles, thus there is definitely no need to link them more than twice, as well as readding/overwriting an existing element already in `Publishers` array.

The warning should be raised to any client sources that mis-initializes the publisher to this very ROS2 node twice, which could help reveal a deeper problem that there might be sth wrong with the Sim initializing process, which technically should only link ROS2Node<->Publisher only once.

Reply:
Well, both `Publisher` and `ROS2Node` are just pointers handles, thus there is definitely no need to link them more than twice, as well as readding/overwriting an existing element already in `Publishers` array.

The warning should be raised to any client sources that mis-initializes the publisher to this very ROS2 node twice/ That could help reveal a deeper problem that there might be sth wrong with the Sim initializing process, which technically should link ROS2Node<->Publisher only once.

Reply:
I see, if there is a case to overwrite existing publisher with different publisher, 
`Node.Publisher[key] = NewPublisher` should work sincd Publishers is able to be edite from bp/cpp.